store/ordersapp/views.py

```diff
@@ -1,3 +1,45 @@
-from django.shortcuts import render
+from datetime import datetime
 
-# Create your views here.
+from django.shortcuts import render, get_object_or_404
+
+from .models import Order, User
+
+
+def get_orders_by_user_id(request, user_id):
+    user = get_object_or_404(User, pk=user_id)
+    orders = Order.objects.filter(user=user)
+    context = {"user": user, "orders": orders}
+    return render(request, "ordersapp/user_orders.html", context)
+
+
+def get_ordered_products_by_user_id(request, user_id):
+    user = get_object_or_404(User, pk=user_id)
+    orders = Order.objects.filter(user=user)
+    result_set_week = set()
+    result_set_month = set()
+    result_set_year = set()
+    today = datetime.now()
+
+    for order in orders:
+        date_diff = (today - datetime.fromisoformat(str(order.date))).days
+        if date_diff <= 7:
+            for product in order.products.all():
+                result_set_week.add(product.title)
+                result_set_month.add(product.title)
+                result_set_year.add(product.title)
+        elif date_diff <= 30:
+            for product in order.products.all():
+                result_set_month.add(product.title)
+                result_set_year.add(product.title)
+        elif date_diff <= 365:
+            for product in order.products.all():
+                result_set_year.add(product.title)
```

Comment:
Я говорил на семинаре, что продукты за 7 дней должны входить в продукты на 30 дней и за 365 дней, значит `elif `тут не подходит
```suggestion
        if date_diff <= 30:
            for product in order.products.all():
                result_set_month.add(product.title)
                result_set_year.add(product.title)
        if date_diff <= 365:
            for product in order.products.all():
                result_set_year.add(product.title)
```